packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -597,30 +598,87 @@ function isDropDownChoice(item: QuestionnaireItem): boolean {
   );
 }
 
+/**
+ * This method is not called `areSameType` because it only checks if the two value's types are not explicitly different and allows one value to be `undefined`.
+ * @param value1 A value or `undefined`.
+ * @param value2 Another value to compare `value1`'s type to.
+ * @returns If the two values are NOT different types. It returns `true` if both are the same type or if `value1` is `undefined`.
+ */
+function areNotDifferentTypes(value1: TypedValue | undefined, value2: TypedValue): boolean {
+  // We return true for `undefined` since we handle the undefined case inside the case blocks
+  if (value1 === undefined || value1.type === value2.type) {
+    return true;
+  }
+  return false;
+}
+
 export function isQuestionEnabled(
   item: QuestionnaireItem,
   answers: Record<string, QuestionnaireResponseItemAnswer>
 ): boolean {
   if (!item.enableWhen) {
     return true;
   }
+
   const enableBehavior = item.enableBehavior ?? 'any';
+
   for (const enableWhen of item.enableWhen) {
-    const expectedAnswer = getTypedPropertyValue(
-      {
-        type: 'QuestionnaireItemEnableWhen',
-        value: enableWhen,
-      },
-      'answer[x]'
-    );
     const actualAnswer = getTypedPropertyValue(
       {
         type: 'QuestionnaireResponseItemAnswer',
         value: answers[enableWhen.question as string],
       },
       'value[x]'
-    );
-    const match = deepEquals(expectedAnswer, actualAnswer);
+    ) as TypedValue | undefined; // possibly undefined when question unanswered
+
+    const expectedAnswer = getTypedPropertyValue(
+      {
+        type: 'QuestionnaireItemEnableWhen',
+        value: enableWhen,
+      },
+      'answer[x]'
+    ) as TypedValue;
+
+    let match: boolean;
+    // We handle exists separately since its so different in terms of comparisons than the other mathematical operators
+    if (enableWhen.operator === 'exists') {
+      // if actualAnswer is not undefined, then exists: true passes
+      // if actualAnswer is undefined, then exists: false passes
+      match = !!actualAnswer === expectedAnswer.value;
+    } else if (!areNotDifferentTypes(actualAnswer, expectedAnswer)) {
+      match = false;
+    } else {
+      switch (enableWhen.operator) {
```

Comment:
The comparison semantics for these values are defined in the [FHIRPath spec](https://hl7.org/fhirpath/#conversion); rather than reimplementing those in this component, we should use [`evalFhirPathTyped()`](https://github.com/medplum/medplum/blob/c7291e67ad636c03379d0c6a58fc12a7086b29f5/packages/core/src/fhirpath/parse.ts#L256) from `@medplum/core` to evaluate expressions like `%actual <= %expected` with the values passed in as `actual` and `expected` variables

Reply:
Okay I made this change but it seems my tests for `valueCoding` when display names are missing or don't match. I'll check around in that `evalFhirPathTyped()` function and see what might be causing that.

Reply:
Thanks for pointing this out by the way, this is way cleaner 👍